src/ServiceControl/SagaAudit/GetSagaByIdApi.cs

```diff
@@ -0,0 +1,37 @@
+namespace ServiceControl.SagaAudit
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Nancy;
+    using Raven.Client;
+    using ServiceControl.CompositeViews.Messages;
+
+    public class GetSagaByIdApi : ScatterGatherApi<Guid, SagaHistory>
+    {
+        public override async Task<QueryResult<SagaHistory>> LocalQuery(Request request, Guid sagaId, string instanceId)
+        {
+            using (var session = Store.OpenAsyncSession())
+            {
+                RavenQueryStatistics stats;
+                var sagaHistory = await 
+                    session.Query<SagaHistory, SagaDetailsIndex>()
+                        .Statistics(out stats)
+                        .SingleOrDefaultAsync(x => x.SagaId == sagaId)
+                        .ConfigureAwait(false);
+
+                if (sagaHistory == null)
+                {
+                    return QueryResult<SagaHistory>.Empty(instanceId);
+                }
+
+                return new QueryResult<SagaHistory>(sagaHistory, instanceId, new QueryStatsInfo(stats.IndexEtag, stats.TotalResults));
+            }
+        }
+
+        protected override SagaHistory ProcessResults(Request request, QueryResult<SagaHistory>[] results)
+        {
+            return results.Select(p => p.Results).FirstOrDefault(x => x != null);
```

Comment:
This has the potential to hide problems. Should this be SingleOrDefault? After all, there should only ever be 0 or 1 result with data. If there's more than 1 then we have a problem.